codegen-client/src/main/kotlin/software/amazon/smithy/rust/codegen/client/smithy/generators/ClientEnumGenerator.kt

```diff
@@ -81,6 +82,10 @@ data class InfallibleEnumType(
 
     override fun additionalEnumMembers(context: EnumGeneratorContext): Writable = writable {
         docs("`$UnknownVariant` contains new variants that have been added since this code was generated.")
+        rust(
+            """##[deprecated(note = "Don't directly match on `$UnknownVariant`. Instead, match on `_` and use \
```

Comment:
I think this is easier if you match on `other if other.as_str() == "Other" { ... }` personally